src/Common.cpp

```diff
@@ -113,6 +118,11 @@ void NyquistIO::Load(AudioData * data, const std::vector<uint8_t> & buffer)
         {
             ext = match_ogg_subtype(buffer.data());
         }
+
+        if (ext != no_extension)
+        {
+            break;
+        }
```

Comment:
ext comparisons could be minimized by nesting

```cpp
for (auto & filetype : magic_map)
{
    if (match_magic(buffer.data(), filetype.first))
    {
        ext = filetype.second;
        if (ext == "ogg_or_vorbis")
        {
            ext = match_ogg_subtype(buffer.data());
        }
        if (ext != no_extension)
        {
              break;
        }
}
```

Reply:
I've reworked it as suggested.

Reply:
Thanks! looks good.